Buttplug.Server.Managers.ErosTek/ETSerialManager.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO.Ports;
+using System.Threading;
+using Buttplug.Core;
+using JetBrains.Annotations;
+
+namespace Buttplug.Server.Managers.ETSerialManager
+{
+    public class ETSerialManager : DeviceSubtypeManager
+    {
+        private bool _isScanning = false;
+        private Thread _scanThread;
+        private string[] _searchComPorts;
+
+        public ETSerialManager(IButtplugLogManager aLogManager)
+            : base(aLogManager)
+        {
+            BpLogger.Info("Loading ErosTek Serial Port Manager");
+        }
+
+        public override void StartScanning()
+        {
+            BpLogger.Info("Starting Scanning Serial Ports for ErosTek Devices");
+            _scanThread = new Thread(() => ScanSerialPorts(_searchComPorts));
+            _scanThread.Start();
+        }
+
+        public override void StopScanning()
+        {
+            BpLogger.Info("Stopping Scanning Serial Ports for ErosTek Devices");
+            _isScanning = false;
+        }
+
+        public override bool IsScanning()
+        {
+            return _isScanning;
+        }
+
+        private void ScanSerialPorts(string[] selectedComPorts)
+        {
+            string[] comPortsToScan;
+            _isScanning = true;
+
+            while (_isScanning)
+            {
+                // Enumerate Ports
+                if (selectedComPorts == null || selectedComPorts.Length == 0)
+                {
+                    comPortsToScan = SerialPort.GetPortNames();
+                }
+                else
+                {
+                    comPortsToScan = selectedComPorts;
+                }
+
+                // try to detect devices on all port
+                foreach (string port in comPortsToScan)
```

Comment:
you can use `var` instead of `string` here to help make your life easier when you decide to refactor later on